File: python/pages/basePage.py
import unittest
from selenium.webdriver.support.ui import Select, WebDriverWait
from selenium.common.exceptions import NoSuchElementException, TimeoutException

class BasePage(unittest.TestCase):

    def _element_present(self, locator):
        try:
            self.driver.find_element_by_css_selector(locator)
        except NoSuchElementException, e: return False
        return True

    def _wait_for(self, locator):
        WebDriverWait(self.driver, 5).until(lambda driver : self._element_present(locator))

File: ruby/pages/base_page.rb
module Pages
  class BasePage

    attr_reader :driver

    def initialize(driver)
      @driver = driver
    end

    def element_present?(locator)
      begin
        find(locator)
      rescue NoSuchElementException
        false
      end
    end

    def wait_for(locator)
      Selenium::WebDriver::Wait.new(:timeout => 5).until { element_present?(locator) }
    end

    def find(locator)
      driver.find_element(css: locator)
    end

    def type(locator, text)
      find(locator).send_keys(text)
    end

    def submit(locator)
      find(locator).submit
    end

    def visit(url_path)
      driver.get(base_url + url_path)
    end

  end
end
